src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/NuGet/NuGetRestorer.cs

```diff
@@ -114,16 +114,19 @@ protected override Task DisposeCoreAsync(bool initialized)
             }
         }
 
-        private async Task ProjectPropertyChangedAsync(Tuple<ImmutableList<IProjectValueVersions>, TIdentityDictionary> sources)
+        private Task ProjectPropertyChangedAsync(Tuple<ImmutableList<IProjectValueVersions>, TIdentityDictionary> sources)
         {
             IVsProjectRestoreInfo projectRestoreInfo = ProjectRestoreInfoBuilder.Build(sources.Item1);
 
             if (projectRestoreInfo != null)
             {
-                await _solutionRestoreService
-                    .NominateProjectAsync(_projectVsServices.Project.FullPath, projectRestoreInfo, CancellationToken.None)
-                    .ConfigureAwait(false);
+                _projectVsServices.Project.Services.ProjectAsynchronousTasks
+                    .RegisterCriticalAsyncTask(JoinableFactory.RunAsync(() => _solutionRestoreService
```

Comment:
NuGet will return the same task if a restore operation is in progress. It also batches restore operations in a queue, so multiple duplicate calls would return together.